doc/development/plugins.rst

```diff
@@ -384,18 +394,55 @@ The user can then import this operation directly from your plugin, and use it wh
     @qnode(dev1)
     def my_qfunc(phi):
         qml.Hadamard(wires=0)
-        CustomGate(phi, wires=[0,1])
+        CustomGate(phi, theta, wires=0)
         return qml.expval(qml.PauliZ(0))
 
 .. warning::
 
     If you are providing custom operations not natively supported by PennyLane, it is recommended
-    that the plugin unittests **do** provide tests to ensure that PennyLane returns the correct
+    that the plugin unit tests **do** provide tests to ensure that PennyLane returns the correct
     gradient for the custom operations.
 
+Supporting new observables
+--------------------------
+
+Custom observables can be added in an identical manner to operations above, but with three
+small changes:
+
+* The :class:`~.Observable` class should instead be subclassed.
+
+* The class attribute :attr:`~.Observable.eigvals` should be defined, returning
+  the eigenvalues of the observable.
+
+* The method :meth:`~.Observable.diagonalizing_gates` should be defined. This method
+  returns a list of PennyLane :class:`~.Operation` objects that diagonalize the observable
+  in the computational basis. This is used to support devices that can only perform
+  measurements in the computational basis.
+
+For example:
+
+.. code-block:: python
 
-Supporting new CV operations
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    class CustomObservable(Observable):
+        """Custom observable"""
+        num_params = 0
+        num_wires = 1
+        par_domain = None
+        eigvals = np.array([0.2, 0.1])
+
+        def diagonalizing_gates(self):
+            return [PauliX(wires=self.wires), Hadamard(wires=self.wires)]
+
+        @staticmethod
+        def _matrix(*params):
+            return np.array([[0, 1], [1, 0]]) / np.sqrt(2)
+
+
+:html:`<div class="note admonition" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content1" class="collapsed"><p class="first admonition-title">CV devices and operations (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content1" class="collapse" data-parent="#aside1" style="height: 0px;">`
+
+**Note: CV devices currently subclass from the base** :class:`~.Device` **class. However, this
+class is deprecated, and a new** ``CVDevice`` **class will be available in the next major
+PennyLane release.**
```

Comment:
Better: will be available soon? I learnt to not promise things if not absolutely necessary :)